modules/apps/product-navigation/product-navigation-site-administration/src/main/resources/META-INF/resources/sites/site_administration_body.jsp

```diff
@@ -95,12 +95,28 @@ Group group = siteAdministrationPanelCategoryDisplayContext.getGroup();
 			</c:if>
 
 			<c:if test="<%= !group.isDepot() && !group.isCompany() %>">
+
+				<%
+				PortletURL portletURL = PortletURLFactoryUtil.create(request, ProductNavigationProductMenuPortletKeys.PRODUCT_NAVIGATION_PRODUCT_MENU, RenderRequest.RENDER_PHASE);
+
+				portletURL.setParameter("mvcPath", "/portlet/pages_tree.jsp");
+				portletURL.setParameter("redirect", themeDisplay.getURLCurrent());
+				portletURL.setParameter("selPpid", portletDisplay.getId());
+				portletURL.setWindowState(LiferayWindowState.EXCLUSIVE);
+				%>
+
 				<clay:button
+					additionalProps='<%=
+						HashMapBuilder.<String, Object>put(
```

Comment:
Same here... the fact that you have to change specific buttons means this change is not safe.

What's causing this issue? If it happens in these 2 places, chances are this happens elsewhere

Reply:
So what I'm seeing is that the "click" event listener for this button is [here](https://github.com/liferay/liferay-portal/blob/b5a4d4c3545feaa2b0682dc43c3985ff3feccbd6/modules/apps/product-navigation/product-navigation-site-administration/src/main/resources/META-INF/resources/sites/site_administration_body.jsp#L132),  then the "React" button is rendered and we "loose" the event listener.

Reply:
it also seems that the `id` is not passed to the React component

Reply:
Why is `react` rendering without `additionalProps` or `propsTransformer`?

Reply:
> Why is react rendering without additionalProps or propsTransformer?

Not sure I expressed myself correctly, 

What I mean is that there's a race condition with the "click" event listener code that's executed before the React button is rendered -  I can use `Liferay.componentReady` to avoid that

Reply:
> Why is react rendering without additionalProps or propsTransformer?

In this pr we're hydrating the button component (by creating the compoonent `Button.js`), so before this one the react component was not being rendered.

Now that it is, as @julien said, we face a possible race condition with code that is adding an event listener to a DOM element, which is being executed before the react component is rendered, so the DOM element where the listener is being attached was the one rendered by JAVA.

Using `Liferay.componentReady` to avoid the problem LGTM. Other option could be not hydrate the component if no `additionalProps` or `propsTransformer` are passed to the tag, but I feel that could cause other issues (talking without testing anything, just a feeling)